simplesat/sat/policy.py

```diff
@@ -13,7 +20,99 @@ def get_next_package_id(self, assignments, clauses):
         """
 
 
+class PolicyLogger(IPolicy):
```

Comment:
After fiddling around with it, I'm not sure that this is better. By wrapping policies rather than the other way around, we make it easy to catch everything on the way in and out and all of the code related to logging is consolidated into this one class.